include/pagmo/problems/luksan_vlcek1.hpp

```diff
@@ -0,0 +1,228 @@
+            f[i + 1] = (3. * std::pow(x[i + 1], 3.) + 2. * x[i + 2] - 5.
+                        + std::sin(x[i + 1] - x[i + 2]) * std::sin(x[i + 1] + x[i + 2]) + 4. * x[i + 1]
+                        - x[i] * std::exp(x[i] - x[i + 1]) - 3.);
+        }
+        return f;
+    }
+    /// Box-bounds
+    /**
+     *
+     * It returns the box-bounds for this UDP.
+     *
+     * @return the lower and upper bounds for each of the decision vector components
+     */
+    std::pair<vector_double, vector_double> get_bounds() const
+    {
+        vector_double lb(m_dim, -5.);
+        vector_double ub(m_dim, 5.);
+        return {lb, ub};
+    }
+    /// Inequality constraint dimension
+    /**
+     *
+     * It returns the number of inequality constraints
+     *
+     * @return the number of inequality constraints
+     */
+    vector_double::size_type get_nec() const
+    {
+        return (m_dim - 2);
+    }
+    /// Gradients
+    /**
+     * It returns the fitness gradient.
+     *
+     * The gradient is represented in a sparse form as required by
+     * problem::gradient().
+     *
+     * @param x the decision vector.
+     *
+     * @return the gradient of the fitness function
+     */
+    vector_double gradient(const vector_double &x) const
+    {
+        assert(x.size() == m_dim);
+        auto n = x.size();
+        vector_double grad(n + 3 * (n - 2), 0.);
+        for (decltype(n) i = 0u; i < n - 1; ++i) {
+            grad[i] += 400. * x[i] * (x[i] * x[i] - x[i + 1]) + 2. * (x[i] - 1.);
+            grad[i + 1] = -200. * (x[i] * x[i] - x[i + 1]);
+        }
+        for (decltype(n) i = 0u; i < n - 2; ++i) {
+            // x[i]
+            grad[n + 3 * i] = -(1. + x[i]) * std::exp(x[i] - x[i + 1]);
+            // x[i+1]
+            grad[n + 1 + 3 * i] = 9. * x[i + 1] * x[i + 1]
+                                  + std::cos(x[i + 1] - x[i + 2]) * std::sin(x[i + 1] + x[i + 2])
+                                  + std::sin(x[i + 1] - x[i + 2]) * std::cos(x[i + 1] + x[i + 2]) + 4.
+                                  + x[i] * std::exp(x[i] - x[i + 1]);
+            // x[i+2]
+            grad[n + 2 + 3 * i] = 2. - std::cos(x[i + 1] - x[i + 2]) * std::sin(x[i + 1] + x[i + 2])
+                                  + std::sin(x[i + 1] - x[i + 2]) * std::cos(x[i + 1] + x[i + 2]);
+        }
+        return grad;
+    }
+    /// Gradients sparsity
+    /**
+     *
+     * It returns the gradent sparisty structure for the Luksan Vlcek 1 problem
+     *
+     * The gradients sparisty is represented in the form required by
+     * problem::gradient_sparsity().
+     *
+     * @return the gradient sparsity structure of the fitness function
+     */
+    sparsity_pattern gradient_sparsity() const
+    {
+        sparsity_pattern retval;
+        // The part relative to the objective function is dense
+        for (decltype(m_dim) i = 0u; i < m_dim; ++i) {
+            retval.push_back({0, i});
```

Comment:
If you want you can write these push_backs as emplace_back without the need of curly brackets.